x/market/handler/endblock.go

```diff
@@ -55,6 +56,49 @@ func transferFundsForActiveLeases(ctx sdk.Context, keepers Keepers) error {
 	return nil
 }
 
+func pickBidWinner(bids []types.Bid) (winner types.Bid) {
+	// open bids; match by lowest price; sort bids by price
+	sort.Slice(bids, func(i, j int) bool {
+		// AFAICT: the BidID DSeq is pulled from the original OrderID.
+		// So it can't be used to determine who bid first.
+		return bids[i].Price.IsLT(bids[j].Price)
+	})
+	if len(bids) == 0 {
+		// This is a fatal case
+		winner = types.Bid{}
+		return
+	} else if len(bids) == 1 {
+		winner = bids[0]
+		return
+	} // else > 2 bids
+
+	if bids[0].Price.IsEqual(bids[1].Price) {
```

Comment:
invert this - do a `!` here and return early.

Reply:
done